c_glib/plasma-glib/client.cpp

```diff
@@ -122,10 +267,160 @@ GPlasmaClient *
+                  "%s Arrow GPU GLib is needed to use GPU",
+                  context);
+      return NULL;
+#endif
+      device_number = options_priv->gpu_device + 1;
+    }
+  }
+  std::shared_ptr<arrow::Buffer> plasma_data;
+  auto status = plasma_client->Create(plasma_id,
+                                      data_size,
+                                      raw_metadata,
+                                      raw_metadata_size,
+                                      &plasma_data,
+                                      device_number);
+  if (garrow_error_check(error, status, context)) {
+    GArrowBuffer *data = nullptr;
+    if (device_number == 0) {
+      auto plasma_mutable_data =
+        std::static_pointer_cast<arrow::MutableBuffer>(plasma_data);
+      data = GARROW_BUFFER(garrow_mutable_buffer_new_raw(&plasma_mutable_data));
+#ifdef HAVE_ARROW_GPU
+    } else {
+      auto plasma_cuda_data =
+        std::static_pointer_cast<arrow::gpu::CudaBuffer>(plasma_data);
+      data = GARROW_BUFFER(garrow_gpu_cuda_buffer_new_raw(&plasma_cuda_data));
+#endif
+    }
+    GArrowBuffer *metadata = nullptr;
+    if (raw_metadata_size > 0) {
+      auto plasma_metadata =
+        std::make_shared<arrow::Buffer>(raw_metadata, raw_metadata_size);
+      metadata = garrow_buffer_new_raw(&plasma_metadata);
+    }
+    return gplasma_created_object_new_raw(client,
+                                          id,
+                                          data,
+                                          metadata,
+                                          device_number - 1);
+  } else {
+    return NULL;
+  }
+}
+
+/**
+ * gplasma_client_refer_object:
+ * @client: A #GPlasmaClient.
+ * @id: The ID of the target object.
+ * @timeout_ms: The timeout in milliseconds. -1 means no timeout.
+ * @error: (nullable): Return location for a #GError or %NULL.
+ *
+ * Returns: (nullable) (transfer full): A found #GPlasmaReferredObject
+ *   on success, %NULL on error.
+ *
+ * Since: 0.12.0
+ */
+GPlasmaReferredObject *
+gplasma_client_refer_object(GPlasmaClient *client,
+                            GPlasmaObjectID *id,
+                            gint64 timeout_ms,
+                            GError **error)
+{
+  const auto context = "[plasma][client][refer-object]";
+  auto plasma_client = gplasma_client_get_raw(client);
+  auto plasma_id = gplasma_object_id_get_raw(id);
+  std::vector<plasma::ObjectID> plasma_ids;
+  plasma_ids.push_back(plasma_id);
+  std::vector<plasma::ObjectBuffer> plasma_object_buffers;
+  auto status = plasma_client->Get(plasma_ids,
+                                   timeout_ms,
+                                   &plasma_object_buffers);
+  if (garrow_error_check(error, status, context)) {
+    auto plasma_object_buffer = plasma_object_buffers[0];
+    auto plasma_data = plasma_object_buffer.data;
+    auto plasma_metadata = plasma_object_buffer.metadata;
+    GArrowBuffer *data;
+    GArrowBuffer *metadata;
```

Comment:
Do we need to silence the warning?
https://travis-ci.org/apache/arrow/jobs/461631864#L3955

Reply:
Yes.
I forgot `return NULL` after `g_set_error()`.